lib/chef/provider/windows_task.rb

```diff
@@ -194,26 +196,56 @@ def run_schtasks(task_action, options = {})
-            @current_resource.cwd != new_resource.cwd ||
-            @current_resource.frequency_modifier != new_resource.frequency_modifier ||
-            @current_resource.frequency != new_resource.frequency ||
-            @current_resource.idle_time != new_resource.idle_time ||
-            @current_resource.random_delay != new_resource.random_delay ||
-            @current_resource.execution_time_limit != new_resource.execution_time_limit ||
-            !new_resource.start_day.nil? || !new_resource.start_time.nil?
+            current_resource.command != new_resource.command.tr("'", '"')) ||
+            current_resource.user != new_resource.user ||
+            current_resource.run_level != new_resource.run_level ||
+            current_resource.cwd != new_resource.cwd ||
+            current_resource.frequency_modifier != new_resource.frequency_modifier ||
+            current_resource.frequency != new_resource.frequency ||
+            current_resource.idle_time != new_resource.idle_time ||
+            current_resource.random_delay != new_resource.random_delay ||
+            !new_resource.execution_time_limit.include?(current_resource.execution_time_limit) ||
+            (new_resource.start_day && start_day_updated?) ||
+            (new_resource.start_time && start_time_updated?)
         begin
-          return true if new_resource.day.to_s.casecmp(@current_resource.day.to_s) != 0 ||
-              new_resource.months.to_s.casecmp(@current_resource.months.to_s) != 0
+          return true if new_resource.day.to_s.casecmp(current_resource.day.to_s) != 0 ||
+              new_resource.months.to_s.casecmp(current_resource.months.to_s) != 0
         rescue
           Chef::Log.debug "caught a raise in task_needs_update?"
         end
 
         false
       end
 
+      def start_day_updated?
+        current_day = to_date_time_obj(current_resource.start_day)
+        new_day = to_date_time_obj(new_resource.start_day)
+        current_day != new_day
+      end
+
+      def start_time_updated?
+        time = to_date_time_obj(current_resource.start_time).strftime("%H:%M")
+        time != new_resource.start_time
+      end
+
+      def to_date_time_obj(date_in_string)
+        return DateTime.parse(date_in_string)
+      rescue
+        require "american_date"
+        return DateTime.parse(date_in_string)
+      end
+
+      def convert_user_date_to_system_date(date_in_string)
+        Chef::Log.debug "Converting '#{date_in_string}' to system date format"
+        date_in_string = to_date_time_obj(date_in_string).strftime("%Y-%m-%d")
+        task_script = <<-EOH
+          [Console]::OutputEncoding = [Text.UTF8Encoding]::UTF8
+          ([datetime]"#{date_in_string}").ToString((((((([Globalization.Cultureinfo]::CurrentCulture.DateTimeFormat.ShortDatePattern -replace "dd", "d") -replace "d", "dd") -replace "MM", "M")-replace "M","MM") -replace "yyyy","yy") -replace "y", "yy"))
```

Comment:
What's the value of doing the replacements in this line.  Can we not use something like the following:

```
C:\Users\spreston> ([datetime]"2017-09-06").ToString([Globalization.Cultureinfo]::CurrentCulture.DateTimeFormat.ShortDatePattern)
06/09/2017
```

Reply:
OK I see it myself, you're trying to protect against dates that have a single character in each section.  Such as 9/6/17.